aiida_vasp/parsers/parser_settings.py

```diff
@@ -73,6 +73,8 @@ def __init__(self, settings, default_settings=None):
 
         self._settings = settings
```

Comment:
To make this a bit more generic I would suggest:
```
    def __init__(self, settings, default_settings=None):

        if settings is None:
            settings = {}
        self._settings = settings
        if default_settings:
            self.update_with(default_settings)

        self.nodes = []
        self.set_nodes()

        self.parser_definitions = {}
        self.set_parser_definitions(self._settings.get('file_parser_set'))
```

Reply:
Are you sure we do not need to preserve the None for the passed settings? That is why I did it like this. In case we need to use the passed settings somewhere else which expects None. Can you check this and I will update the PR.

Reply:
At the moment the only other place, where settings is used afterwards is in the `VasprunParser` and there 'None' is not required.

Also python works a bit different when passing arguments to a function. It does not always mean that a reference is passed. If you assign a new instance of an object to a parameter of a function that will not influence the original object that has been passed to the function. Here is the corresponding example:
```
In [1]: def function1(my_dict):
   ...:     my_dict['a'] = 1
   ...:     
In [2]: def function2(my_dict):
   ...:     my_dict = {}
   ...:     
In [3]: d = {}
In [4]: print d
{}
In [5]: function1(d)
In [6]: print d
{'a': 1}
In [7]: function2(d)
In [8]: print d
{'a': 1}
```

So overall I am actually fine with both solutions.